### DISCUSSION TITLE
Upload a load profile to the platform

### DISCUSSION THREAD

**Discussion by MidrarAdham** [GRIDAPPSD/gridappsd-forum#149] [Q&A] [answered]
Hello Everyone,

I'm trying to upload a load profile containing Watts values to the platform so it can be read by the EnergyConsumers later instead of the ieeezipload.glm. Is there a way to do this?

Thank you

**Comment by poorva1209** [comment, marked as answer]
Yes, you can upload a custom load profile.

Assuming you have a load profile file in this format https://github.com/GRIDAPPSD/gridappsd-data/blob/master/timeseries/loadprofiles/ieeezipload.player, you can follow the steps mentioned here: https://github.com/GRIDAPPSD/gridappsd-data/tree/master/timeseries/loadprofiles to upload them to the timeseries data store which can then be used by the simulation. 

The instructions are written for someone familiar with GridAPPS-D docker containers and InfluxDB. Let me know if you need more details with them.

**Reply by MidrarAdham** [reply]
I will. Thank you, Poorva!

**Reply by MidrarAdham** [reply]
Hello Poorva,

I ran the simulation, and it looks like the old ieeezipload.glm is generated again. 

Here are the steps I followed to generate the new load profile:

- I loaded the influxdb container.

- I adjusted the ieeezipload.player file as in the [above repo link](https://github.com/GRIDAPPSD/gridappsd-data/tree/master/timeseries/loadprofiles ).

    - I also tried different branches (master, develop, release/2023.05.0, and upload_update)

- I executed the loadprofile_measurement_bulk.py, which generated loadprofile_measurement_out.txt.

    - I checked the generated file, and the values match the input file.
  
- I executed the ```docker cp loadprofile_measurement_out.txt influxdb:/``` , which generated the following output:

    - ```Successfully copied 17.9MB to influxdb:/```
    - I also checked the influxdb container and can see the file in the container's home directory.
   
- I executed the ```influx -import -path=loadprofile_measurement_out.txt -precision s```, which generated the following output:

```
2023/07/11 14:22:29 Processed 100000 lines.  Time elapsed: 802.039789ms.  Points per second (PPS): 124682
2023/07/11 14:22:30 Processed 200000 lines.  Time elapsed: 1.566315075s.  Points per second (PPS): 127688
2023/07/11 14:22:30 Processed 300000 lines.  Time elapsed: 2.323791446s.  Points per second (PPS): 129099
2023/07/11 14:22:31 Processed 400000 lines.  Time elapsed: 3.079578854s.  Points per second (PPS): 129887
2023/07/11 14:22:32 Processed 500000 lines.  Time elapsed: 3.825765431s.  Points per second (PPS): 130692
2023/07/11 14:22:32 Processed 1 commands
2023/07/11 14:22:32 Processed 525600 inserts
2023/07/11 14:22:32 Failed 0 inserts
```

- I also tried uploading the file to the /tmp/ folder in the influxdb container, but still the older version of ieeezipload.player gets generated during the simulation.

Thank you

**Comment by MidrarAdham** [comment]
Hi Poorva,

I got it resolved. I'm going to post the solution here for people who may encounter the same issue. 

What caused the issue before is that I changed the ```seed_epoch_date``` value in the loadprofile_measurement_bulk.py. Once I put it back as it was, the ieeezipload.player within the gridappsd container is updated correctly.

original value: ```seed_epoch_date = 1514764800```

Here is the commands sequence I used:

- Clone the repo in [Poorva's post above](https://github.com/GRIDAPPSD/gridappsd-data/tree/master/timeseries/loadprofiles )

- load the influxdb container.

- update the ieeezipload.player.

- Execute ```docker cp loadprofile_measurement_out.txt influxdb:/```

- Execute ```docker exec -t influxdb influx -import -path=loadprofile_measurement_out.txt -precision s```

- Then run the list_measurement and insert_measurement scripts to the desired feeder model.

Thank you for the help, Poorva!
